simulation/navigator_gazebo/test/test_sim_integration.py

```diff
@@ -0,0 +1,161 @@
+
+    def odom_cb(self, msg):
+        self.odom_pos_msg.append(msg.pose.pose.position.x)
+        self.odom_pos_msg.append(msg.pose.pose.position.y)
+        self.odom_pos_msg.append(msg.pose.pose.position.z)
+        self.odom_ori_msg.append(msg.pose.pose.orientation.x)
+        self.odom_ori_msg.append(msg.pose.pose.orientation.y)
+        self.odom_ori_msg.append(msg.pose.pose.orientation.z)
+        self.odom_ori_msg.append(msg.pose.pose.orientation.w)
+
+    def test_odom(self):
+        timeout = rospy.Time.now() + rospy.Duration(1)
+        while ((self.odom_pos_msg == False or self.odom_ori_msg == False)
+               and rospy.Time.now() < timeout):
+            rospy.Sleep(0.01)
+        self.assertTrue(self.odom_pos_msg and self.odom_ori_msg)
+        initial_pos = [-1.2319, 0.0, 0.0]
+        initial_ori = [0.0, 0.0, 0.0, 1.0]
+        self.verify_pos_ori(
+            self.odom_pos_msg,
+            initial_pos,
+            self.odom_ori_msg,
+            initial_ori,
+            "/odom")
+
+    def absodom_cb(self, msg):
+        self.absodom_pos_msg.append(msg.pose.pose.position.x)
+        self.absodom_pos_msg.append(msg.pose.pose.position.y)
+        self.absodom_pos_msg.append(msg.pose.pose.position.z)
+        self.absodom_ori_msg.append(msg.pose.pose.orientation.x)
+        self.absodom_ori_msg.append(msg.pose.pose.orientation.y)
+        self.absodom_ori_msg.append(msg.pose.pose.orientation.z)
+        self.absodom_ori_msg.append(msg.pose.pose.orientation.w)
+
+    def test_absodom(self):
+        timeout = rospy.Time.now() + rospy.Duration(1)
+        while ((self.absodom_pos_msg == False or self.absodom_ori_msg ==
+                False) and rospy.Time.now() < timeout):
+            rospy.Sleep(0.01)
+        self.assertTrue(self.absodom_pos_msg and self.absodom_ori_msg)
+        initial_pos = [743789.637462, -5503821.62581, 3125622.25266]
+        initial_ori = [0.0, 0.0, 0.0, 1.0]
+        self.verify_pos_ori(
+            self.absodom_pos_msg,
+            initial_pos,
+            self.absodom_ori_msg,
+            initial_ori,
+            "/absodom")
+
+    def verify_pos_ori(self, pos, initial_pos, ori, initial_ori, topic):
+        # make assertions
+        for actual, initial in itertools.izip(pos, initial_pos):
+            self.assertAlmostEqual(
+                actual, initial, places=4, msg=(
+                    "Error: {} position is: {} should be {}".format(
+                        topic, actual, initial)))
+        for actual, initial in itertools.izip(ori, initial_ori):
+            self.assertEqual(
+                actual, initial, msg=(
+                    "Error: {} orientation is: {} should be {}".format(
+                        topic, actual, initial)))
+
+    def cam_info_cb(self, msg):
+        self.cam_info_msg.append([msg.width, msg.height])
+
+    def cam_image_cb(self, msg):
+        self.cam_image_msg.append(msg.data)
+
+    def test_image(self):
```

Comment:
AFAIK this will not check that each image arrived, but only check the validity of the ones that arrived. You need to check the length of the camera responses